calico-cloud/image-assurance/understanding-scan-results.mdx

```diff
@@ -34,6 +34,7 @@ Other notes:
 - The columns, **Clusters** and **Running Instances**, show the number of running instances in clusters that are connected to {{prodname}}.
 - The **Unknown** scan result filter reflects images that are not fully scanned. Because they can add noise to the table, they are disabled by default. To enable Unknown results for strategic troubleshooting, click the **Result** drop-down menu and select **Unknown**.
 - In the All Scanned Images and Running Images tabs, the **Registry path** field may be blank if {{prodname}} cannot access this metadata. For example, images from Docker Hub do not specify the registry in the image metadata.
+- Any exceptions configured for image scanning will be applicable to Runtime View as well. 
```

Comment:
@penkeysuresh I'm worried this is too small of a call out for a user to notice. 

If we expect all of our customers to be trained on Image Assurance through customer success, then this isn't a problem. 

I do recognize that we don't really have a dedicated section on the Runtime view, so this section is the best that we have, if we don't want to start refactoring the docs structure. 

But I do think we need to make this clearer and more explicit. 

By clearer, I mean we should attempt to spell this out more and ensure the user isn't confused. For example, the user could get confused and think this is referring to CVE exceptions that they can create in the UI. 

By explicit, I mean we should try to link this back to the other section that you have created on the cluster scanning page, showing the full definition in the ImageAssurance CR. 

But I think we want to try, as much as possible, to make the user docs self-serve.

Reply:
^yeah, let me know what you have in mind @stevegaossou. I'll add it.

Reply:
> This seems to be mentioned in a note under the section ### Configure exclusions for image scanning.

It is, but that's on a completely different page. 

How is someone how is only interested in Runtime View, supposed to know that? Keep in mind that would have no real insight to check the page for cluster scanning, because all they are interested in is Runtime View. 

My suggestion is that we refer to the other page from this section here and also call it out on this page a bit more explicitly.

Reply:
> My suggestion is that we refer to the other page from this section

Sure, I'll add those. 

> also call it out on this page a bit more explicitly

What do you mean by this ? We are calling it right now. Do you mean we bold the text or something else ?

Reply:
I mean that the current statement is not clear enough on its own to help explain to the user what they need to do to exclude namespaces from Runtime View: 
> Any exceptions configured for image scanning will be applicable to Runtime View as well. 

Here's what I said in my previous statement in this thread: 
"By clearer, I mean we should attempt to spell this out more and ensure the user isn't confused. For example, the user could get confused and think this is referring to CVE exceptions that they can create in the UI."

I think it is still relevant. 

Adding a link reference from this statement to the detailed section on the cluster scanning page helps a lot to clarify to the user what this is talking about. 

If we wanted to make it even clearer, perhaps we could change the wording here. 

For example go from: 

> Any exceptions configured for image scanning will be applicable to Runtime View as well. 

To: 

> You can exclude namespaces from Runtime View by applying the following configuration (URL ref here).

Reply:
This is why defining user stories for docs (at the doc plan stage) is helpful. 

I'd been reading this as a feature for cluster scanning, with a limitation that applies to the runtime view side of things. For that, the docs appear more or less adequate. 

Now I'm thinking you have a different idea, basically two user stories:

As a user, I want to define exceptions ( for the cluster scanner | for runtime view ) so I can eliminate noise in my scan results.

If that's the case, then it's probably useful to write a generic procedure, add it to both pages, and make sure that both call out the fact that the other system is similarly affected.

Is that the right idea?

Reply:
Apologies for the tardiness on replying. I lost track of this thread end of last week. I'm just coming back to it now. 

> This is why defining user stories for docs (at the doc plan stage) is helpful.

Sure, we could talk about the gap and what we could have done better to avoid the misalignment here. Perhaps a Slack thread? I'm fine with whatever channel you think works best. I'm keen to ensure we adjust and improve communication and coordination for user docs. 

You can see the product requirements and context for this feature here: 
- https://tigera.atlassian.net/browse/PMREQ-602

> I'd been reading this as a feature for cluster scanning, with a limitation that applies to the runtime view side of things. For that, the docs appear more or less adequate.

This new feature, which I refer to as namespace exclusion list, affects the Runtime View and Cluster Scanning in equal parts. So it's not really for one or the other. It's for both. 

> As a user, I want to define exceptions ( for the cluster scanner | for runtime view ) so I can eliminate noise in my scan results.

Note, we should avoid calling it exceptions, because we have another feature that uses that term (i.e. CVE exceptions). So I'm trying to avoid confusion in the user docs language by referring to this as the namespace exclusion list. 

> If that's the case, then it's probably useful to write a generic procedure, add it to both pages, and make sure that both call out the fact that the other system is similarly affected.
> Is that the right idea?

I think your suggestion makes sense. Because this namespace exclusion list appears in equal measure to both Runtime View and Cluster Scanning, it probably makes sense to feature the related content equally for both features. 

I think perhaps part of the challenge when Suresh was going through this is that the Runtime View doesn't really have its own section. It's essentially embedded into a broader section about the Running Images tab in the Image Assurance UI. 

@penkeysuresh what do you think? Should we try to have the same level of detail on both pages? That might involve you refactoring a bit the existing section about Running Images tab.

Reply:
I don't have a strong opinion on the layout of the documentation, I think it would be best to leave this decision to you and Phil.

From my perspective, the current documentation is effectively addressing the behaviour of exclusion criteria. I feel like having sections for each feature might give the impression that there are two distinct exclusion lists, which is not the case. Even if you add two different sections you'll end up adding note to both of them saying that they'll apply to both cluster scanner and runtime view.

We have received customer requests to include exclusions for the cluster scanner, so I believe it's appropriate to consolidate the documentation in that section. We can then matter of factly add a note in the runtime view section that the same exclusions apply. If you don't think that's appropriate let me know, I'll make any changes you suggest.

Reply:
I'm fine as it is as long as it's easy to spot on the page for Runtime View. 

Let's leave this open until @WilliamTigera goes through the test cycles for Namespace Exclusion and get his feedback after he reads the user docs. 

I think the restructuring was more a suggestion from Chris. If I were to take that suggestion even further, perhaps what we eventually want to do is have a new page that summaries features in a table format, separate from the guides that we have today. And on that page we could talk about the configuration options for each feature. I haven't fully thought it out, and it's not something that I am suggestion we tackle right now. 

But I think the pages we have currently are more general guides for specific use cases. I think there's benefits to having an alternative documentation where we go into more detail the configuration options for each feature in a kind of directory or listing (similar to the API reference). 

Anyways, @penkeysuresh I think you can leave it as it is now (unless Chris says otherwise). And we'll wait to see if William has any feedback from a user's perspective when he does the test cycles.

Reply:
I think the restructuring to use a common template was more a suggestion from Chris. If I were to take that suggestion even further, perhaps what we eventually want to do is have a new page that summaries features in a table format, separate from the guides that we have today. And on that page we could talk about the configuration options for each feature. I haven't fully thought it out, and it's not something that I am suggestion we tackle right now. 

But I think the pages we have currently are more general guides for specific use cases. I think there's benefits to having an alternative documentation where we go into more detail the configuration options for each feature in a kind of directory or listing (similar to the API reference). 

Anyways, @penkeysuresh I think you can leave it as it is now (unless Chris says otherwise). And we'll wait to see if William has any feedback from a user's perspective when he does the test cycles.